contracts/liquidity_hub/epoch-manager/src/commands.rs

```diff
@@ -27,27 +27,27 @@ pub(crate) fn remove_hook(
 }
 
 /// Creates a new epoch.
-pub fn create_epoch(
-    deps: DepsMut,
-    _env: Env,
-    info: MessageInfo,
-) -> Result<Response, ContractError> {
+pub fn create_epoch(deps: DepsMut, env: Env, info: MessageInfo) -> Result<Response, ContractError> {
     cw_utils::nonpayable(&info)?;
 
     let mut current_epoch = query_current_epoch(deps.as_ref())?.epoch;
     let config = CONFIG.load(deps.storage)?;
-    println!("Creating new epoch");
-
-    // if env
-    //     .block
-    //     .time
-    //     .minus_nanos(current_epoch.start_time.nanos())
-    //     .nanos()
-    //     < config.epoch_config.duration.u64()
-    // {
-    //     return Err(ContractError::CurrentEpochNotExpired);
-    // }
-    println!("Creating new epoch");
+
+    ensure!(
+        env.block.time >= current_epoch.start_time,
+        ContractError::GenesisEpochHasNotStarted
+    );
+
+    if env
+        .block
+        .time
+        .minus_nanos(current_epoch.start_time.nanos())
+        .nanos()
+        < config.epoch_config.duration.u64()
```

Comment:
make sure epoch duration is in nanos too

Reply:
It should be in nanos as the struct suggests.

```
#[cw_serde]
pub struct EpochConfig {
    /// The duration of an epoch in nanoseconds.
    pub duration: Uint64,
    /// Timestamp for the first epoch, in nanoseconds.
    pub genesis_epoch: Uint64,
}
```